server/src/core/api-rest/api-rest-query-builder/factories/parsers/filter-parser.factory.ts

```diff
@@ -0,0 +1,216 @@
+        bracketsCounter--;
+      }
+      if (c === ',' && parenthesisCounter === 1 && bracketsCounter === 0) {
+        predicates.push(currentPredicates);
+        currentPredicates = '';
+        continue;
+      }
+      if (parenthesisCounter >= 1) currentPredicates += c;
+    }
+    if (currentPredicates.length) {
+      predicates.push(currentPredicates);
+    }
+    return predicates;
+  }
+
+  parseSimpleFilterString(filterString: string): {
+    fields: string[];
+    comparator: string;
+    value: string;
+  } {
+    if (!filterString.match(`^(.+)\\[(.+)\\]:(.+)$`)) {
+      throw Error(`'filter' invalid for '${filterString}'. eg: price[gte]:10`);
+    }
+    let fields = '';
+    let comparator = '';
+    let value = '';
+    let previousC = '';
+    let fillFields = true;
+    let fillComparator = false;
+    let fillValue = false;
+    for (const c of filterString) {
+      if (c === '[') {
+        fillFields = false;
+      }
+      if (previousC === '[' && !comparator.length) {
+        fillComparator = true;
+      }
+      if (c === ']') {
+        fillComparator = false;
+      }
+      if (previousC === ']' && c === ':' && !value.length) {
+        fillValue = true;
+        continue;
+      }
+      if (fillFields) {
+        fields += c;
+      }
+      if (fillComparator) {
+        comparator += c;
+      }
+      if (fillValue) {
+        value += c;
+      }
+      previousC = c;
+    }
+    if (!Object.keys(FILTER_COMPARATORS).includes(comparator)) {
+      throw Error(
+        `'filter' invalid for '${filterString}', comparator ${comparator} not in ${Object.keys(
+          FILTER_COMPARATORS,
+        ).join(',')}`,
+      );
+    }
+    return { fields: fields.split('.'), comparator, value };
+  }
+
+  formatFieldValue(value, fieldType?, comparator?) {
+    if (comparator === 'in') {
+      if (value[0] !== '[' || value[value.length - 1] !== ']') {
+        throw Error(
+          `'filter' invalid for 'in' operator. Received '${value}' but array value expected eg: 'field[in]:[value_1,value_2]'`,
+        );
+      }
+      const stringValues = value.substring(1, value.length - 1);
+      return stringValues
+        .split(',')
+        .map((value) => this.formatFieldValue(value, fieldType));
+    }
+    if (comparator === 'is') {
+      return value;
+    }
+    if (fieldType === 'NUMBER') {
+      return parseInt(value);
+    }
+    if (fieldType === 'BOOLEAN') {
+      return value.toLowerCase() === 'true';
+    }
+    if (
+      (value[0] === '"' || value[0] === "'") &&
+      (value.charAt(value.length - 1) === '"' ||
+        value.charAt(value.length - 1) === "'")
+    ) {
+      return value.substring(1, value.length - 1);
+    }
+    return value;
+  }
+
+  parseStringFilter(filterQuery, objectMetadataItem) {
+    const result = {};
+    const match = filterQuery.match(
+      `^(${Object.values(CONJUNCTIONS).join('|')})((.+))$`,
+    );
+    if (match) {
+      const conjunction = match[1];
```

Comment:
@martmull Event if match exist it doesn't necessary mean `match[1]` exist and is accessible.
You should also check if `conjunction` is defined afterward 
```suggestion
      const conjunction = match?.[1];
```